src/raftstore/store/snap.rs

```diff
@@ -254,9 +255,12 @@ mod v1 {
     /// `save` method to make them persistent. When saving a crc32 checksum
     /// will be appended to the file end automatically.
     pub struct Snap {
+        mgr: SnapManager,
```

Comment:
Related code checked, It's OK and it's the saftest way to maintain the snap-ref-status in the snap-mgr, IMO.
BTW, this source file is far too large, it's literally the longest file in the project.